### ISSUE TITLE
vic machine fails install when insecure-registry specified

### ISSUE THREAD

**Issue by cgtexmex** [vmware/vic#4141]
if installing a VCH and you specify an insecure registry the vic-machine parsing of the flag fails.

vic built from source using `go1.8`

vic-machine create command:
`vic-machine-linux create -t 172.16.73.170 -u root -p vagrant -n vickie -vs datastore1/vickie-vols:default -vs datastore1/vickie-vols:jojo -k -f --insecure-registry 172.16.73.147:5000`

vic-machine result:
```
Mar  3 2017 22:59:57.228Z INFO  ### Installing VCH ####
Mar  3 2017 22:59:57.229Z WARN  Using administrative user for VCH operation - use --ops-user to improve security (see -x for advanced help)
Mar  3 2017 22:59:57.229Z WARN  Configuring without TLS - all communications will be insecure
Mar  3 2017 22:59:57.229Z ERROR --------------------
Mar  3 2017 22:59:57.230Z ERROR vic-machine-linux create failed: 172.16.73.147:5000 is an invalid format for registry url
```

This worked with `go version 1.7.x` but fails with `go version 1.8` -- specifically the call to `url.Parse(registry_string)` throws an error.  An example is at:  https://play.golang.org/p/SwZAq5NszD -- this is executing in go1.8, but if you run locally with go1.7x no error will be thrown.

Additionally, if you provide a url that will pass the validation then the pull from the private registry doesn't work.

From my perspective this effectively breaks insecure-registry support.

**Comment by cgtexmex** [comment]
I have compiled vic-machine-linux with `go version go1.7.4 linux/amd64` and I get no parsing error.  

If I compile vic-machine-linux with `go version go1.8 linux/amd64` I get the parsing error.

**Comment by cgtexmex** [comment]
If you supply `//172.16.73.147:5000` to vic-machine then it works as expected.  So we either need to update the documentation or handle this in code.

**Comment by mhagen-vmware** [comment]
I hit this in #4232 as well, if you don't specify the port number I could get it to work. So you need to specify it with just the IP address.  I closed that issue and duped it here.  We need to be able to support port number and protocols in our specification.

**Comment by hickeng** [comment]
@lubronzhan this is the issue you were talking about - however with hostname instead of IP the parse succeeds,

**Comment by hickeng** [comment]
This completely breaks our ability to specify insecure registries - even if a hostname is used instead it gets incorrectly parsed.

**Comment by stuclem** [comment]
I missed this issue until now as I have been filtering by doc milestone. If this is being fixed for 1.1 then we can remove `kind/user-docs`.

Added (belatedly) to the Release Notes for 0.9:

-----

- <b>VCH deployment fails with invalid URL error when using `--insecure-registry`. [#4141](https://github.com/vmware/vic/issues/4141)</b><br>If you use the `--insecure-registry` option with `vic-machine create`, deployment of the VCH fails with the following error:<pre><i>registry_address</i>:5000 is an invalid format for registry url.</pre>

   Workaround: Precede the registry address with double forward slashes.<pre>//<i>registry_address</i>:5000</pre>

-----

@cgtexmex does this cover it?

**Comment by mhagen-vmware** [comment]
This is required to be fixed by 1.1

**Comment by stuclem** [comment]
This is in the resolved issues for 1.1
